frontend/src/refactor/RegionList.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {Component} from "react";
+import http from "../../http.js";
+import {FormControl, Grid, InputLabel, MenuItem, Select} from "@mui/material";
+import RegionListItem from "./RegionListItem.jsx";
+import DeleteConfirmDialog from "./DeleteConfirmDialog.jsx";
+import RegionModal from "./RegionModal.jsx";
+
+/**
+ * Generates a dropdown list or Regions that also manages their deletion
+ */
+class RegionList extends Component{
+
+    /**
+     *
+     * @param props
+     * @param props.selected_region_id:String - the id of the selected region
+     * @param props.select_region:Function - What to do when a region has been selected
+     */
+    constructor(props) {
+        super(props);
+
+        this.delete_region = this.delete_region.bind(this);
+        this.append_region = this.append_region.bind(this);
+
+        this.state = {
+            regions:[],
+            selected_region:"",
+            del_region:{},
+            show_del:false
+        }
+    }
+
+    /**
+     * Fetches the regions from the server and assigns them to the component, selecting the one active for the site
+     * if it exists
+     */
+    componentDidMount() {
+        http.get("/regions")
+            .then(regs => {
+                this.setState({regions: regs.data})
+
+                //If we have been configured with a region, select it.
+                if (this.props.selected_region_id !== undefined) {
+                    let sel_reg = regs.data.find(reg => reg._id === this.props.selected_region_id);
+                    this.select_region(sel_reg)
+                }
+            })
+            .catch(err => console.error("Failed to fetch regions",err));
+    }
+
+    /**
+     * Facilitates opening the deletion dialogue
+     * @param reg:Region - the region to delete
+     */
+    confirm_reg_delete = (reg) => {
+        this.setState({del_region:reg,show_del:true})
+    }
+
+
+    /**
+     * Actually deletes the region of the supplied ID
+     * @param reg_id:String the ID of the region to delete
+     */
+    delete_region = (reg_id) => {
+        http.delete(`/regions/${reg_id}`)
+            .then(() => {
+                console.log(`Successfully deleted region: ${reg_id} `)
+                let regions_updated = this.state.regions.filter(reg => reg._id !== reg_id)
+                this.setState({regions:regions_updated,
+                    del_reg:{},
+                    show_del:false,
+                    selected_region:""//todo this blanks the set region, even if the deleted one is not the set one...
```

Comment:
what is this todo for? It clearly zeros out selected_region, but why is this a problem in a delete function?

Reply:
Ah, i need to assign this to one of the new issues. Basically, if you go to edit a site's region, and decide to delete a region from that modal, it will unset the region for the site you are editing *even if the region you deleted was not the one you had set originally*. I think that is no good. If you deleted the region you had previously set, it would make sense. But that is the only case i see this as permissible.

Reply:
Also, Rockstar review as always :)

Reply:
moved to #2